### PR TITLE
gunicorn Fix for Eventlet

### PR DESCRIPTION
**Pull request by k-macmillan**
# Description
It was identified that there is a [fix for the import error](https://github.com/benoitc/gunicorn/pull/2581). This change locks us to [that point](https://github.com/benoitc/gunicorn/tree/6a8ebb4844b2f28596ffe7421eb9f7d08c8dc4d8) in the repository.

This allows us to be up to date with `eventlet` as well.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran locally
- [x] PR unit tests passed
- [x] [Deployed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/4473069988)
- [x] userflows [passes](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/4473069988/jobs/7860280807)
- [x] email test send

Speedy deploy
image

Sent myself an email:
image

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: requirements-app.txt [hunk 1]
```diff
@@ -23,7 +23,7 @@ docopt>=0.6.2
 # security vulnerabilities 40399 and 40416, which are ignored in Makefile.  The ImportError
 # traceback begins in Gunicorn code, which does not properly use newer eventlet versions.
 #     https://github.com/benoitc/gunicorn/issues?q=is%3Aissue+is%3Aopen+ALREADY_HANDLED
-eventlet==0.30.2
+eventlet~=0.33.3
```
Comment (kalbfled) [inline requirements-app.txt:26]: Note the above comment.  If this works, you will want to remove the cited vulnerabilities from the Makefile and change this comment.

File: requirements-app.txt [hunk 2]
```diff
@@ -43,7 +43,7 @@ git+https://github.com/mitsuhiko/flask-sqlalchemy.git@500e732dd1b975a56ab06a46bd
 # The production image defined by Dockerfile uses Gunicorn.  The development image defined by Dockerfile.local
 # runs the application with the Flask development server by default.
 # 9 May 2022: Newer versions cause build or runtime issues.
-gunicorn[eventlet] < 20.1.0
+gunicorn[eventlet] @ git+https://github.com/benoitc/gunicorn.git@6a8ebb4844b2f28596ffe7421eb9f7d08c8dc4d8
```
Comment (kalbfled) [inline requirements-app.txt:46]: Please document why this needs to be pinned to a specific commit.

File: requirements.txt [hunk 3]
```diff
@@ -5,13 +5,13 @@ asn1crypto==1.5.1
...
-awscli==1.27.94
+awscli==1.27.95
 bcrypt==4.0.1
 billiard==3.6.4.0
 bleach==6.0.0
 blinker==1.5
-boto3==1.26.94
-botocore==1.29.94
+boto3==1.26.95
+botocore==1.29.95
```
Comment (kalbfled) [inline requirements.txt:14]: These are top-level requirements.  Although not strictly necessary, whenever I see top-level updates in requirements.txt, I also update them in requirements-app.txt.  See [step 4](https://github.com/department-of-veterans-affairs/notification-api#update-requirementstxt).

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: requirements-app.txt (hunk 1 above)
Comment (k-macmillan) [reply requirements-app.txt:22]: Done.

File: requirements-app.txt (hunk 2 above)
Comment (k-macmillan) [reply requirements-app.txt:42]: Done.

File: requirements.txt (hunk 3 above)
Comment (k-macmillan) [reply requirements.txt:14]: Added to requirements-app.txt.

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**